packages/jsii/lib/compiler.ts

```diff
@@ -145,9 +159,102 @@ export class Compiler implements Emitter {
                 target: COMPILER_OPTIONS.target && ts.ScriptTarget[COMPILER_OPTIONS.target],
                 jsx: COMPILER_OPTIONS.jsx && Case.snake(ts.JsxEmit[COMPILER_OPTIONS.jsx]),
             },
-            [commentKey]: commentValue
-        }, {  replacer: utils.filterEmpty, spaces: 4 });
-        return configPath;
+            include: ["**/*.ts"],
+            exclude: ["node_modules"],
+            // Change the references a little. We write 'originalpath' to the
+            // file under the 'path' key, which is the same as what the
+            // TypeScript compiler does. Make it relative so that the files are
+            // movable. Not strictly required but looks better.
+            references: references.map(p => ({ path: p })),
+        } as any;
+    }
+
+    /**
+     * Creates a `tsconfig.json` file to improve the IDE experience.
+     *
+     * @return the fully qualified path to the ``tsconfig.json`` file
+     */
+    private async writeTypeScriptConfig(): Promise<void> {
+        const commentKey = '_generated_by_jsii_';
+        const commentValue = 'Generated by jsii - safe to delete, and ideally should be in .gitignore';
+
+        (this.typescriptConfig as any)[commentKey] = commentValue;
+
+        if (await fs.pathExists(this.configPath)) {
+            const currentConfig = await fs.readJson(this.configPath);
+            if (!(commentKey in currentConfig)) {
+                // tslint:disable-next-line:max-line-length
+                throw new Error(`A '${this.configPath}' file that was not generated by jsii is in ${this.options.projectInfo.projectRoot}. Aborting instead of overwriting.`);
+            }
+        }
+        LOG.debug(`Creating or updating ${colors.blue(this.configPath)}`);
+        await fs.writeJson(this.configPath, this.typescriptConfig, {  replacer: utils.filterEmpty, spaces: 4 });
+    }
+
+    /**
+     * Find all dependencies that look like TypeScript projects.
+     *
+     * Enumerate all dependencies, if they have a tsconfig.json file with
+     * "composite: true" we consider them project references.
+     *
+     * Unfortunately it doesn't seem like the TypeScript compiler itself
+     * resolves transitive references in a way that
+     */
+    private async findProjectReferences(): Promise<string[]> {
+        const packageJsonPath = path.join(this.options.projectInfo.projectRoot, 'package.json');
+
+        // If there's no package file, don't do anything
+        if (!await fs.pathExists(packageJsonPath)) { return []; }
+        const pkg = require(packageJsonPath);
```

Comment:
I believe we are loading package.json like 10 times already. If not too hard, maybe we can just pass that into the function?